### PR TITLE
Logging: Drop Settings from some logging ctors

### PR DESCRIPTION
**Pull request by nik9000**
Drops `Settings` from some logging ctors now that they are no longer
needed. This should allow us to stop passing `Settings` around to quite
as many places.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/logging/Loggers.java [hunk 1]
```diff
@@ -49,16 +49,17 @@
         Setting.prefixKeySetting("logger.", (key) -> new Setting<>(key, Level.INFO.name(), Level::valueOf, Setting.Property.Dynamic,
             Setting.Property.NodeScope));
 
-    public static Logger getLogger(Class<?> clazz, Settings settings, ShardId shardId, String... prefixes) {
-        return getLogger(clazz, settings, shardId.getIndex(), asArrayList(Integer.toString(shardId.id()), prefixes).toArray(new String[0]));
+    public static Logger getLogger(Class<?> clazz, ShardId shardId, String... prefixes) {
+        return getLogger(clazz, Settings.EMPTY,
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/common/logging/Loggers.java]: I'll make follow up changes that remove the need for `Settings.EMPTY` eventually.

Comment (nik9000) [reply server/src/main/java/org/elasticsearch/common/logging/Loggers.java]: They are just needed to make it compile and aren't used.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by nik9000** [comment]
Thanks for reviewing @jasontedor!

**Pull request merged**